aips/account_abstraction.md

```diff
@@ -0,0 +1,213 @@
+    },
+}
+```
+where `function_info` indicates the function registered on the `sender` account that will be called as authentication check
+and `signature` is a byte array that includes all the necessary information as the input to the function.
+
+If the `AccountAuthenticator` is not `Abstraction`, the `authentication_key` passed to the prologue will be `Some(auth_key)`;
+while if it is `Abstraction` then `authentication_key == None`
+
+### Dispatchable Authenticator
+
+At the framework level, we create a new resource, `DispatchableAuthenticator`. If an account has `DispatchableAuthenticator`, 
+it allows the account to be authenticated via the functions registered in the internal map.
+
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+/// The dispatchable authenticator that defines how to authenticate this account in the specified module.
+/// An integral part of Account Abstraction.
+struct DispatchableAuthenticator has key, copy, drop {
+    auth_functions: SimpleMap<FunctionInfo, bool>
+}
+
+enum AbstractionAuthData has copy, drop {
+    V1 { digest: vector<u8>, authenticator: vector<u8> },
+}
+
+fun authenticate(account: signer, func_info: FunctionInfo, auth_data: AbstractionAuthData): signer acquires DispatchableAuthenticator {
+    let func_infos = dispatchable_authenticator_internal(signer::address_of(&account));
+    assert!(simple_map::contains_key(func_infos, &func_info), error::not_found(EFUNCTION_INFO_EXISTENCE));
+    function_info::load_module_from_function(&func_info);
+    dispatchable_authenticate(account, signing_data, &func_info)
+}
+
+/// The native function to dispatch customized move authentication function.
+native fun dispatchable_authenticate(account: signer, signing_data: SigningData, function: &FunctionInfo): signer;
+
+```
+
+Any authentication function must have the same interface as the following function:
+```rust
+/// The native function to dispatch customized move authentication function.
+public fun dispatchable_authenticate(account: signer, signing_data: SigningData): signer;
+```
+The first parameter, `signer`, represents the current account that needs authentication, and the function's return value is also a `signer`. Typically, this function returns the input signer if the authentication check is successful. However, in some cases, a permissioned signer may be returned to control the resources the transaction can access.
+`auth_data` is an enum that contains all the data required to authenticate the transaction via account abstraction. The initial version just has two fields, where `digest` is the sha3 digest of the corresponding transaction signing message and `authenticator` is the attached byte array which serves as the proof of the authentication verification. Usually it encodes the signature over the digest at least.
+Best Practice: To prevent replay attacks (both cross-chain), the `authenticator` must depend on both `chain_id` and `digest`.
```

Comment:
isn't chain_id already part of digest, because it is part of RawTransaction?

Reply:
oh yes.